samcli/commands/local/lib/sam_api_provider.py

```diff
@@ -47,9 +42,17 @@ def __init__(self, template_dict, parameter_overrides=None, cwd=None):
         ----------
         template_dict : dict
             SAM Template as a dictionary
+
         cwd : str
             Optional working directory with respect to which we will resolve relative path to Swagger file
+
+        provider_type: AbstractParserProvider
+            Object to parse the api configurations
         """
+        self.provider_type = provider_type or self.PROVIDER_TYPE_CLOUD_FORMATION
```

Comment:
Not sure I understand what the purpose of this is. Why check for `PROVIDER_TYPE_CLOUD_FORMATION` and what does this represent?